datadog_checks_base/datadog_checks/base/utils/warnings_util.py

```diff
@@ -0,0 +1,36 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import warnings
+from contextlib import contextmanager
+
+from six import PY2
+
+
+def _simplefilter_py2(action, category=Warning, lineno=0, append=0):
+    """ Add remove logic for py2 to avoid warnings.filters to growth
+        indefinitely if simplefilter is called multiple times """
+    item = (action, None, category, None, int(lineno))
+    if not append:
```

Comment:
Not sure why you check that.

Reply:
Python uses the same logic here: https://github.com/python/cpython/blob/1f864016950079a618a916140f3abe6480a8e22d/Lib/warnings.py#L184-L189

I think it's good to have the same behaviour for py2 and py3.

Reply:
Added a test for this case: https://github.com/DataDog/integrations-core/pull/5080/commits/9bc997f3749d7d4fc1d92ef9ffe7d3adc9303f81